modules/contacts/client/components/RemoveContact.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withNamespaces } from 'react-i18next';
+import { Modal } from 'react-bootstrap';
+
+export function RemoveContact({ t, contact, show, inProgress, onHide, onRemoveContact, situation='confirmed' }) {
+
+  // Different confirm button label and modal title depending on situation
+  const situationLabels = {
+    // User is cancelling a request they sent
+    unconfirmedFromMe: {
+      confirm: 'Yes, revoke request',
+      title: 'Revoke contact request?',
+      time: 'Requested'
+    },
+    // Decline received request
+    unconfirmedToMe: {
+      confirm: 'Yes, decline request',
+      title: 'Decline contact request?',
+      time: 'Requested'
+    },
+    // Removing confirmed contact
+    confirmed: {
+      confirm: 'Yes, remove contact',
+      title: 'Remove contact?',
+      time: 'Connected since'
```

Comment:
Shouldn't these be wrapped with `t()` already here instead of `t(labels.title)`?

Otherwise parsers won't be able to pick these strings.

Reply:
:+1: 
I'll change that.

FYI: There is also a [runtime extraction](https://react.i18next.com/guides/extracting-translations#3-runtime-extraction), which, when properly set, could save to some api whatever keys were missing. In that context this would work, too. It's on my wishlist todo.

Reply:
Neat! Probably kinda early to look into stuff like that. :-) For now we should spread good patterns everyone can copy across the codebase.

Reply:
15f8cba5a99783deab9b254792a76a43bb56f99b